### PR TITLE
Fix: fix sync gitee failed & solve warning problems in action.

### PR DESCRIPTION
**Pull request by mo3et**
This PR is solve sync gitee failed problem. Refer to [go-kratos/workfows/gitee-sync.yml](https://github.com/go-kratos/kratos/blob/main/.github/workflows/gitee-sync.yml#L17).

### CODE REVIEW

**Comment by mo3et** [comment]
Other commit solve the warning problems. Such it ([detail](https://github.com/go-kratos/kratos-layout/actions/runs/8075051402)).

image

If dont create `Setup Environment` and `Module Cache`. `Setup-Go` will print error.

```
Restore cache failed: Dependencies file is not found in /home/runner/work/kratos-layout/kratos-layout. Supported file pattern: go.sum
```

It's change also refer to [kratos workflows config](https://github.com/go-kratos/kratos/blob/main/.github/workflows/go.yml#L56)

**Comment by mo3et** [comment]
Could you please review this PR？
@Windfarer

File: .github/workflows/go.yml [hunk 1]
```diff
@@ -2,35 +2,48 @@ name: Go
...
+          if [ -f Gopkg.toml ]; then
+              curl https://raw.githubusercontent.com/golang/dep/master/install.sh | sh
+              dep ensure
+          fi
+
+      - name: Build
+        run: go build -v ./...
+
+      - name: Test
+        run: go test -v ./...
```
Comment (demoManito) [inline .github/workflows/go.yml:49]: No need to indent two spaces

https://yaml.org/spec/1.2.2/#21-collections

Comment (mo3et) [reply .github/workflows/go.yml]: Latest commit is fixed. Thank you!

Comment (demoManito) [reply .github/workflows/go.yml]: Hello, there is still one more space, just keep it at the same level as `steps`
image

Comment (mo3et) [reply .github/workflows/go.yml]: Sorry, my editor display is not obvious. Now fixed it.

**Review by demoManito — APPROVED** [review]

**Pull request merged**